### PR TITLE
[hip][tests] add a unit test for testing hipLaunchCooperativeKernel

### PR DESCRIPTION
**Pull request by asalmanp**

### CODE REVIEW

File: tests/src/runtimeApi/module/hipLaunchCooperativeKernel.cpp [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+    if (threadIdx.x == 0) {
+        sum = 0;
+        for (uint i = 0; i < blockDim.x; i++) {
+            sum += tmp[i];
+        }
+        tmpBuf[blockIdx.x] = sum;
+    }
+
+    // TODO replace this line with the sync function once it's available
+    __ockl_gws_barrier(nwm1, 0);
```
Comment (b-sumner) [inline tests/src/runtimeApi/module/hipLaunchCooperativeKernel.cpp]: Note that this function does not guarantee that all the updates to tmpBuf will be visible when the barrier is passed.  If this produces the expected sum, it is just luck.

File: tests/src/runtimeApi/module/hipLaunchCooperativeKernel.cpp [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+    params[1] = (void*)&BufferSizeInDwords;
+    params[2] = (void*)&dB;
+    params[3] = (void*)&dC;
+
+    std::cout << "Testing with grid size = " << dimGrid.x << " and block size = " << dimBlock.x << "\n";
+    HIPCHECK(hipLaunchCooperativeKernel(test_gws, dimGrid, dimBlock, params, dimBlock.x * sizeof(long), stream));
+
+    HIPCHECK(hipMemcpy(init, dC, sizeof(long), hipMemcpyDeviceToHost));
+
+    if (*dC != (((long)(BufferSizeInDwords) * (BufferSizeInDwords - 1)) / 2)) {
```
[the remark this answers is not in the document]
Comment (asalmanp) [reply tests/src/runtimeApi/module/hipLaunchCooperativeKernel.cpp]: The test should be fine in the sense that it can help to test the new API and catch some potential regressions. There are rooms for improving the test to catch some corner cases and make it stronger. These improvements and adding more tests will be done in follow up PRs.

**Comment by asalmanp** [comment]
Ping!
